spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/DatabaseClientTransactionManagerAutoConfiguration.java

```diff
@@ -0,0 +1,63 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.autoconfigure.spanner;
+
+import com.google.cloud.spanner.DatabaseClient;
+
+import org.springframework.boot.autoconfigure.AutoConfigureOrder;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnClass;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnSingleCandidate;
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.cloud.gcp.data.spanner.core.SpannerTemplate;
+import org.springframework.cloud.gcp.data.spanner.core.SpannerTransactionManager;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.Ordered;
+import org.springframework.transaction.annotation.EnableTransactionManagement;
+
+/**
+ * Auto-configuration for {@link SpannerTransactionManager}.
+ *
+ * @author Chengyuan Zhao
+ *
+ * @since 1.1
+ */
+@Configuration
+@EnableTransactionManagement
+@ConditionalOnClass({ SpannerTemplate.class, SpannerTransactionManager.class })
```

Comment:
Both of these classes are in the same module. You should only need one of them.
Can we also make it conditional on a class in `spring-tx`? Of course, `spring-tx` would need to be an `optional` dependency. I remember you saying something before that `spring-tx` was already needed for something non-transactional in our Spanner implementation. Can you look into that?

Reply:
Will make the conditional-on-class changes as requested.

About spring-tx, it was added because our exception type `SpannerDataException` extends `org.springframework.dao.DataAccessException`

Reply:
noted